lib/Frontend/FrontendInputsAndOutputs.cpp

```diff
@@ -45,6 +46,7 @@ operator=(const FrontendInputsAndOutputs &other) {
   for (InputFile input : other.AllInputs)
     addInput(input);
   IsSingleThreadedWMO = other.IsSingleThreadedWMO;
+  LinesToExecute = other.LinesToExecute;
```

Comment:
My bad! The duplication between line 40 and 49 makes my failure to factor `FrontendInputsAndOutputs::FrontendInputsAndOutputs(const FrontendInputsAndOutputs &other)` with `FrontendInputsAndOutputs &FrontendInputsAndOutputs::operator=(const FrontendInputsAndOutputs &other)` stick out like a sore thumb. @jrose-apple do you have a good suggestion for how to factor those duplicated 4 lines? I guess a private static function could work.

Reply:
Since we're not too concerned about performance, it's usually possible to implement the copy-constructor as

```cxx
FrontendInputsAndOutputs::FrontendInputsAndOutputs(const FrontendInputsAndOutputs &other) {
  *this = other;
}
```

But if we did care about the cost of default initialization, then there's really no way to do this. C++ just doesn't provide it.